dd-sdk-android-sqldelight/src/main/kotlin/com/datadog/android/sqldelight/SqlDelightExt.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sqldelight
+
+import com.datadog.android.tracing.AndroidTracer
+import com.squareup.sqldelight.Transacter
+import com.squareup.sqldelight.Transacter.Transaction
+import com.squareup.sqldelight.TransactionWithReturn
+import com.squareup.sqldelight.TransactionWithoutReturn
+import io.opentracing.Span
+import io.opentracing.util.GlobalTracer
+
+// region Public API
+
+/**
+ * Starts a [Transaction] and runs [body] in that transaction.
+ * A span will be created around the transaction code and sent to Datadog.
+ *
+ * @param operationName the name of the [Span] created around the coroutine code.
+ * @param noEnclosing in case we want the currently opened transaction to be automatically closed
+ * @param body the code to be executed inside the transaction
+ * @throws IllegalStateException if [noEnclosing] is true and there is already an active
+ *   [Transaction] on this thread.
+ */
+fun <T : Transacter> T.transactionTraced(
+    operationName: String,
+    noEnclosing: Boolean = false,
+    body: TransactionWithSpanAndWithoutReturn.() -> Unit
+) {
+    withinSpan(operationName, GlobalTracer.get().activeSpan()) {
+        transaction(noEnclosing = noEnclosing) {
+            body.invoke(TransactionWithSpanAndWithoutReturnImpl(this@withinSpan, this))
+        }
+    }
+}
+
+/**
+ * Starts a [Transaction] and runs [body] in that transaction.
+ * A span will be created around the transaction code and sent to Datadog.
+ *
+ * @param operationName the name of the [Span] created around the coroutine code.
+ * @param noEnclosing in case we want the currently opened transaction to be automatically closed
+ * @param body the code to be executed inside the transaction
+ * @throws IllegalStateException if [noEnclosing] is true and there is already an active
+ *   [Transaction] on this thread.
+ */
+fun <T : Transacter, R> T.transactionTracedWithResult(
+    operationName: String,
+    noEnclosing: Boolean = false,
+    body: TransactionWithSpanAndWithReturn<R>.() -> R
+): R {
+    withinSpan(operationName, GlobalTracer.get().activeSpan()) {
+        return transactionWithResult(noEnclosing = noEnclosing) {
+            body.invoke(TransactionWithSpanAndWithReturnImpl(this@withinSpan, this))
+        }
+    }
+}
+
+/**
+ * An object that implements both [Span] and [TransactionWithReturn].
+ */
+interface TransactionWithSpanAndWithReturn<R> : TransactionWithReturn<R>, Span
```

Comment:
This is nitpicking but I'd rather see that interface in its own file, like I did for coroutines